File: backend/src/main/java/team16/spring_project1/domain/order/Mapper/OrderMapper.java
package team16.spring_project1.domain.order.Mapper;

import org.springframework.stereotype.Component;
import team16.spring_project1.domain.order.DTO.response.OrderItemResponseDTO;
import team16.spring_project1.domain.order.DTO.response.OrderResponseDTO;
import team16.spring_project1.domain.order.Entity.Order;
import team16.spring_project1.domain.order.Entity.OrderItem;

import java.util.List;

@Component
public class OrderMapper {

    public OrderResponseDTO toOrderResponseDTO(Order order) {
		order.getStatus();
		order.getOrderItems();
		return new OrderResponseDTO(
                order.getId(),
                order.getEmail(),
			order.getStatus().name(),
                order.getTotalPrice(),
                order.getCreateDate(),
                order.getModifyDate(),
			order.getOrderItems().stream().map(this::toOrderItemResponseDTO).toList()
        );
    }

    public OrderItemResponseDTO toOrderItemResponseDTO(OrderItem item) {
        return new OrderItemResponseDTO(
                item.getId(),
                item.getProductName(),
                item.getCount(),
                item.getPrice(),
                item.getCreateDate(),
                item.getModifyDate()
        );
    }

}

File: backend/src/main/java/team16/spring_project1/domain/order/Mapper/OrderMapper.kt
package team16.spring_project1.domain.order.Mapper

import org.springframework.stereotype.Component
import team16.spring_project1.domain.order.DTO.response.OrderItemResponseDTO
import team16.spring_project1.domain.order.DTO.response.OrderResponseDTO
import team16.spring_project1.domain.order.Entity.Order
import team16.spring_project1.domain.order.Entity.OrderItem

@Component
class OrderMapper {
    fun toOrderResponseDTO(order: Order): OrderResponseDTO {
        order.status
        order.orderItems
        return OrderResponseDTO(
            order.id,
            order.email,
            order.status.name,
            order.totalPrice,
            order.createDate,
            order.modifyDate,
            order.orderItems.stream().map { item: OrderItem -> this.toOrderItemResponseDTO(item) }.toList()
        )
    }

    fun toOrderItemResponseDTO(item: OrderItem): OrderItemResponseDTO {
        return OrderItemResponseDTO(
            item.id!!,
            item.productName,
            item.count,
            item.price,
            item.createDate!!,
            item.modifyDate!!
        )
    }
}
